Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications_test.py

```diff
@@ -137,3 +144,266 @@ def test_service_principal_list_command(mocker, requests_mock,
     assert get_mock.call_count == list_url_count
     assert next_link_get_mock.call_count == next_url_count
     assert ('?$top=' in str(get_mock.last_request)) == top_param_in_url
+
+
+@pytest.mark.parametrize('args, expected_args',
+                         [
+                             ({'id': 'TEST', 'app_id': None}, "/TEST"),
+                             ({'id': None, 'app_id': 'TEST'}, "(appId='TEST')"),
+                             ({'id': 'TEST', 'app_id': 'TEST'}, "/TEST")
+                         ])
+def test_remove_service_principals_command(mocker, requests_mock, my_client, args, expected_args):
+    """
+        Given:
+            - Required arguments (id or app_id or both)
+        When:
+            - Executing msgraph-apps-service-principal-remove command
+        Then:
+            - Ensure delete_service_principals client's method was called with the right argument
+    """
+    from MicrosoftGraphApplications import remove_service_principals_command
+
+    mock_token = {'access_token': 'test_token', 'expires_in': '86400'}
+    requests_mock.post('https://login.microsoftonline.com/organizations/oauth2/v2.0/token', json=mock_token)
+    requests_mock.delete("https://graph.microsoft.com/v1.0/servicePrincipals/TEST", json={})
+    mock_delete_service_principals = mocker.patch.object(my_client, "delete_service_principals")
+    remove_service_principals_command(my_client, args=args)
+
+    assert mock_delete_service_principals.call_args[0][0] == expected_args
+
+
+def test_remove_service_principals_command_validation(requests_mock, my_client):
+    """
+        Given:
+            - No arguments were given
+        When:
+            - Executing msgraph-apps-service-principal-remove command
+        Then:
+            - Ensure the validation works as expected and raise an exception to missing arguments
+    """
+    from MicrosoftGraphApplications import remove_service_principals_command
+    mock_token = {'access_token': 'test_token', 'expires_in': '86400'}
+    requests_mock.post('https://login.microsoftonline.com/organizations/oauth2/v2.0/token', json=mock_token)
```

Comment:
do we still these lines now when we have the fixture?

Reply:
Removed them when they weren't needed